cluster/saltbase/salt/kube-master-addons/alarm

```diff
@@ -0,0 +1,56 @@
+#!/bin/sh
+
+# Copyright 2014 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Execute a command with a timeout
+# Usage: alarm timeout_in_seconds command
+
+cleanup() {
+    # Reset handler to default
+    trap - ALRM
+    # Stop timer subshell if running
+    kill -ALRM $watcher_pid 2>/dev/null
+    # Kill last job
+    kill -9 $! 2>/dev/null &&
+    # exit with 124 if it was running
+    exit 124
+}
+
+watchit() {
+    trap "cleanup" ALRM
+    sleep $1& wait
```

Comment:
Ahh, I realized the reason why I put background and wait here last night. Without this one, if command terminated earlier, I still need to wait until timeout, which is not acceptable.